array-manipulation-1/src/compact.ts

```diff
@@ -0,0 +1,20 @@
+/* exported compact */
+function compact(array: unknown[]): unknown[] {
+  const compactArr = [];
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] !== false) {
+      if (array[i] !== null) {
+        if (!Number.isNaN(array[i])) {
+          if (array[i] !== 0) {
+            if (array[i] !== '') {
+              if (array[i] !== undefined) {
+                compactArr.push(array[i]);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  return compactArr;
+}
```

Comment:
We can actually simplify this! `false`, `null`, etc are referred to as "falsy". This means that in the context of a conditional expression, they will always evaluate to `false`. So this specific problem is checking to see if a value evaluates to either `true` or `false`. Since all conditional expressions evaluate to either `true` or `false`, we can just do this

```
function compact(array: unknown[]): unknown[] {
  const output = [];
  for (let i = 0; i < array.length; i++) {
    const element = array[i];
    if (element) output.push(element);
  }
  return output;
}
```